sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/impl/GrpcStubFactory.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.sgv2.restsvc.impl;
+
+import io.stargate.proto.StargateGrpc.StargateBlockingStub;
+import io.stargate.sgv2.restsvc.resources.AuthenticationFilter;
+import javax.inject.Inject;
+import javax.ws.rs.container.ContainerRequestContext;
+import org.glassfish.hk2.api.Factory;
+
+/**
+ * Provides the stub that we created in {@link AuthenticationFilter} to the Jersey context, so that
+ * it can be injected into resource methods with {@code @Context}.
+ */
+public class GrpcStubFactory implements Factory<StargateBlockingStub> {
```

Comment:
This is slightly more convoluted than I'd like.
I wish there was a way to register the stub in the request scope directly from the filter, but it seems like we need this additional factory.

Reply:
Quick question: since `Factory` comes from Glassfish, does this mean that the registration mechanism is Jersey-specific? Which would mean it is one more thing to port over to another framework if we choose to move to something like Quarkus.

Reply:
Yes, I think it's implementation-specific.
With that being said, other Glassfish types are already referenced in `RestServiceServer`, so there will be a small porting effort no matter what.

Reply:
Agreed.